osx/2_build_toolchain.sh

```diff
@@ -44,17 +44,22 @@ function set_build_flags {
 	CLANGXX=`xcodebuild -find clang++`
 	ARCH="-arch x86_64"
 	SDKPATH=`xcrun -sdk macosx --show-sdk-path`
+	SYSROOT="-mmacosx-version-min=10.9 -isysroot $SDKPATH"
 
 	export CC="$CLANG $ARCH"
 	export CXX="$CLANGXX $ARCH"
 	if [ "$ENABLE_CCACHE" ]; then
 		export CC="ccache $CC"
 		export CXX="ccache $CXX"
 	fi
-	export CFLAGS="-g -O2 -mmacosx-version-min=10.9 -isysroot $SDKPATH"
-	export CXXFLAGS=$CFLAGS
-	export CPPFLAGS="-I$PLATFORM_PREFIX/include"
-	export LDFLAGS="-L$PLATFORM_PREFIX/lib $ARCH -mmacosx-version-min=10.9 -isysroot $SDKPATH"
+	export CFLAGS="-g -O2 $SYSROOT"
+	export CPPFLAGS="-I$PLATFORM_PREFIX/include $SYSROOT"
```

Comment:
CXXFLAGS -> CPPFLAGS is needed for libpng. Otherwise doesn't use -isysroot

Reply:
Well, `CPPFLAGS` is the right place for `-I` stuff anyway.

Reply:
Can you try moving this down to `install_lib` call for png?
`configure` will add environment flags if passed as parameter, so should work and looks more clean to me.